Sources/Common/Core/Math/index.js

```diff
@@ -757,23 +766,28 @@ export function jacobiN(a, n, w, v) {
   let tmp;
   const b = createArray(n);
   const z = createArray(n);
+  const a_temp = createArray(n * n);
+  for (let index = 0; index < n * n; index++) {
+    a_temp[index] = a[index];
+  }
 
-  const vtkROTATE = (aa, ii, jj, kk, ll) => {
-    g = aa[ii][jj];
-    h = aa[kk][ll];
-    aa[ii][jj] = g - s * (h + g * tau);
-    aa[kk][ll] = h + s * (g - h * tau);
+  const vtkROTATE = (aa, ii, jj) => {
+    g = aa[ii];
+    h = aa[jj];
+    aa[ii] = g - s * (h + g * tau);
+    aa[jj] = h + s * (g - h * tau);
   };
 
   // initialize
-  for (ip = 0; ip < n; ip++) {
-    for (iq = 0; iq < n; iq++) {
-      v[ip][iq] = 0.0;
+  for (ip = 0; ip < n * n; ip++) {
+    v[ip] = 0.0;
+    if (ip % (n + 1) === 0) {
+      v[ip] = 1.0;
     }
-    v[ip][ip] = 1.0;
+    // v[ip] = ip % (n + 1) === 0 ? 1.0 : 0.0;
   }
```

Comment:
previous implementation was more elegant.

(remove comment)

you could create an identity function that takes n as parameter

Reply:
I think last commit solves this problem.